controllers/controllers.go

```diff
@@ -287,3 +296,39 @@ func markAsFreeOnSuccess(c client.Client, watchNamespace string, recorder record
 	}
 	return err
 }
+
+// markNodeAsUnhealthy marks the given node as unhealthy by adding an annotation to it. If the number of unhealthy nodes
+// exceeds the maximum allowed, an error is returned.
+func markNodeAsUnhealthy(c client.Client, node *core.Node) error {
+	nodeLocker.Lock()
+	defer nodeLocker.Unlock()
+
+	ctx := context.TODO()
```

Comment:
we should start passing context's into functions rather than using context.TODO everywhere, please add it as a parameter for `markNodeAsUnhealthy`

Reply:
+1

Reply:
True and no context is available from the caller block. So it is basically moving the `context.TODO()` to the function call. Is that what you mean? I don't think I can afford the level of refactoring to also make `ensureInstanceIsUpToDate()` context aware, at least in this PR.

Reply:
Yep all I'm asking is for the addition of context to the parameters of this one function.
Not asking for a refactor.

Reply:
ACK